### PR TITLE
FOUNDATIONS: Add Event

### LINKED ISSUE #259: FOUNDATIONS: Add Event
Dependent on #254

### PR DESCRIPTION
**Pull request by khusnii**
closes #259

### CODE REVIEW

File: Taarafo.Core.Tests.Unit/Services/Foundations/Events/EventServiceTests.Logic.Add.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            //given
+            DateTimeOffset randomDate = GetRandomDateTimeOffset();
+            Event randomEvent = CreateRandomEvent();
+            Event inputEvent = randomEvent;
+            Event storageEvent = inputEvent;
+            Event expectedEvent = storageEvent.DeepClone();
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.InsertEventAsync(inputEvent))
+                .ReturnsAsync(storageEvent);
```
Comment (Javlonbek-dev) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Events/EventServiceTests.Logic.Add.cs:30]: Tab

Comment (Javlonbek-dev) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Events/EventServiceTests.Logic.Add.cs:30]: Tab here

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Hi @khusnii, you need to add the validation and exception tests for Add.

File: Taarafo.Core.Tests.Unit/Services/Foundations/Events/EventServiceTests.Logic.Add.cs (hunk 1 above)
Comment (Javlonbek-dev) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Events/EventServiceTests.Logic.Add.cs:30]: Tab here

File: Taarafo.Core.Tests.Unit/Services/Foundations/Events/EventServiceTests.Exceptions.Add.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                broker.InsertEventAsync(It.IsAny<Event>()), Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionAs(
+                    expectedEventDependencyException))), Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+    }
```
Comment (cjdutoit) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Events/EventServiceTests.Exceptions.Add.cs:54]: Can you also add these tests:
`public async Task ShouldThrowDependencyValidationExceptionOnAddIfEventAlreadyExsitsAndLogItAsync()`

`public async Task ShouldThrowDependencyExceptionOnAddIfDatabaseUpdateErrorOccursAndLogItAsync()`

`public async Task ShouldThrowServiceExceptionOnAddIfServiceErrorOccursAndLogItAsync()`

**Comment by cjdutoit** [comment]
@khusnii  going to close this PR as it's inactive.

**Pull request closed without merging**